openqa-clone-and-monitor-job-from-pr

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env perl
+
+# Note that this script is intended for execution via the GitHub action defined in
+# actions/clone-job/action.yaml.
+
+package openqa_clone_and_monitor_job_from_pr;
+
+use Mojo::Base -strict, -signatures;
+use Mojo::JSON qw(decode_json);
+
+my $expected_url = $ENV{OPENQA_HOST} // 'https://openqa.opensuse.org';
+my $group_id = $ENV{OPENQA_SCHEDULE_GROUP_ID} // 118;
+my $pr_body = $ENV{GH_PR_BODY} or die 'GH_PR_BODY must be set';
+my $gh_repo = $ENV{GH_REPO} or die 'GH_REPO must be set';
+my $gh_ref = $ENV{GH_REF} or die 'GH_REF must be set';
+my $gh_srv = $ENV{GITHUB_SERVER_URL} or die 'GITHUB_SERVER_URL must be set';
+my $api_key = $ENV{OPENQA_API_KEY} or die 'OPENQA_API_KEY must be set';
+my $api_secret = $ENV{OPENQA_API_SECRET} or die 'OPENQA_API_SECRET must be set';
+my $screts = "--apikey '$api_key' --apisecret '$api_secret'";
+my $vars = "BUILD='$gh_repo.git#$gh_ref' _GROUP_ID='$group_id' CASEDIR='$gh_srv/$gh_repo.git#$gh_ref'";
+
+sub _parse_urls ($text) {
+    my @urls;
+    while ($text =~ /(\@openqa:?\s+Clone\s+(https?:[^\s]+))/ig) {
+        push @urls, $2 if index($2, $expected_url) == 0;
+    }
+    return \@urls if @urls;
+    print 'No test cloned; the PR description does not contain ';    # uncoverable statement
+    print "a command like '\@openqa: Clone $expected_url/tests/<JOB_ID>'.\n";    # uncoverable statement
+    exit 0;
+}
+
+sub _run_cmd ($name, $args, $capture_output = 0) {
+    my $res = $capture_output ? `$name $args` : system "$name $args";
```

Comment:
We should probable make sure that the arguments cannot be used to run arbitrary commands, not sure if this would be possible with the current inputs. But better be safe than sorry.
If this is used for pull_request_target, it could be possible to leak our API credentials.

Reply:
We *could* but we needed to add `OpenQA::` to the search path. And we needed to make sure not to break the API in the future.

Reply:
hm, now I'm wondering why the openQA modules aren't installed in the usual place for perl modules, where it would be found by default.
I understand that for the os-autoinst modules that could be a bad idea, but OpenQA is a clear namespace.

Reply:
```
open(my $fh, "-|", $cmd, @args);
while (<$fh>) { ... }
```

Reply:
And to clarify what I meant and what we discussed in the call: If a user manages to manipulate any of the input variables to contain anything then the secrets can be leaked: `'; curl-command-to-post-secrets-to-some-server; '`